File: CareConnect/CareConnect.Models/SearchObjects/WorkshopSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class WorkshopSearchObject : BaseSearchObject<WorkshopAdditionalData>
    {
        public string? NameGTE { get; set; } = null!;

        public string? Status { get; set; } = null!;

        public DateTime? DateGTE { get; set; }

        public DateTime? DateLTE { get; set; }

        public decimal? Price { get; set; }

        public int? MaxParticipants { get; set; }

        public int? Participants { get; set; }

        public string? WorkshopType { get; set; }

        public int? ParticipantId { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/search_objects/workshop_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'workshop_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class WorkshopSearchObject {
  String? fts;
  final String? nameGTE;
  final String? status;
  final DateTime? dateGTE;
  final DateTime? dateLTE;
  final double? price;
  final int? maxParticipants;
  final int? participants;
  final String? workshopType;
  final int? participantId;
  int? page;
  String? sortBy;
  bool? sortAscending;
  bool? includeTotalCount;
  bool? retrieveAll;

  WorkshopSearchObject({
    this.fts,
    this.nameGTE,
    this.status,
    this.dateGTE,
    this.dateLTE,
    this.price,
    this.maxParticipants,
    this.participants,
    this.workshopType,
    this.participantId,

    this.page,
    this.sortBy,
    this.sortAscending,
    this.includeTotalCount,
    this.retrieveAll,
  });

  factory WorkshopSearchObject.fromJson(Map<String, dynamic> json) =>
      _$WorkshopSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$WorkshopSearchObjectToJson(this);
}
